samples/javascript_nodejs/09.message-routing/dialogs/greeting/greeting.js

```diff
@@ -125,61 +122,60 @@ class Greeting extends ComponentDialog {
+        const greetingState = await this.greetingStateAccessor.get(step.context);
         if (greetingState.city === undefined && step.result) {
             let lowerCaseCity = step.result;
             // capitalize and set city
             greetingState.city = lowerCaseCity.charAt(0).toUpperCase() + lowerCaseCity.substr(1);
-            await this.greetingStateAccessor.set(dc.context, greetingState);
-            await this.userState.saveChanges(dc.context);
+            await this.greetingStateAccessor.set(step.context, greetingState);
+            await this.userState.saveChanges(step.context);
         }
-        return await this.greetUser(dc);
+        return await this.greetUser(step);
     }
     /**
      * Validator function to verify that user name meets required constraints.
-     * 
-     * @param {DialogContext} context for this dialog
+     *
      * @param {PromptValidatorContext} prompt context for this prompt
      */
-    async validateName(context, prompt) {
+    async validateName(prompt) {
         // Validate that the user entered a minimum lenght for their name
+        prompt.activeDialog = await this.greetingStateAccessor;
+        prompt.context = await prompt.context;
         const value = (prompt.recognized.value || '').trim();
         if (value.length >= NAME_LENGTH_MIN) {
-            prompt.end(value);
+            return true;
         } else {
-            await context.sendActivity(`Names need to be at least ${NAME_LENGTH_MIN} characters long.`);
+            await prompt.context.sendActivity(`Names need to be at least ${NAME_LENGTH_MIN} characters long.`);
         }
     }
     /**
      * Validator function to verify if city meets required constraints.
-     * 
-     * @param {DialogContext} context for this dialog
+     *
      * @param {PromptValidatorContext} prompt context for this prompt
      */
-    async validateCity(context, prompt) {
+    async validateCity(prompt) {
         // Validate that the user entered a minimum lenght for their name
         const value = (prompt.recognized.value || '').trim();
         if (value.length >= CITY_LENGTH_MIN) {
-            prompt.end(value);
+            return true;
         } else {
-            await context.sendActivity(`City names needs to be at least ${CITY_LENGTH_MIN} characters long.`);
+            await prompt.context.sendActivity(`City names needs to be at least ${CITY_LENGTH_MIN} characters long.`);
         }
     }
     /**
      * Helper function to greet user with information in greetingState.
-     * 
+     *
      * @param {DialogContext} dc context for this dialog
      */
     async greetUser(dc) {
         const greetingState = await this.greetingStateAccessor.get(dc.context);
         // Display to the user their profile information and end dialog
         await dc.context.sendActivity(`Hi ${greetingState.name}, from ${greetingState.city}, nice to meet you!`);
-        return await dc.end();
+        return await dc.endDialog();
```

Comment:
This function takes a `WaterfallStepContext`, not a `DialogContext` instance, please change the code and docstrings for this function.